hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/EightToSevenDowngradeHandler.java

```diff
@@ -92,4 +115,52 @@ private static void downgradePartitionFields(HoodieWriteConfig config, HoodieEng
       tablePropsToAdd.put(HoodieTableConfig.PARTITION_FIELDS, tableConfig.getPartitionFieldProp());
     }
   }
+
+  static void downgradeMetadataPartitions(HoodieEngineContext context,
+                                          HoodieStorage hoodieStorage,
+                                          HoodieTableMetaClient mdtMetaClient,
+                                          Map<ConfigProperty, String> tablePropsToAdd) {
+    // Fetch metadata partition paths.
+    List<String> metadataPartitions = FSUtils.getAllPartitionPaths(
+        context,
+        hoodieStorage,
+        mdtMetaClient.getBasePath(),
+        true);
+
+    // Delete partitions.
+    List<String> validPartitionPaths = deleteMetadataPartition(context, mdtMetaClient, metadataPartitions);
+
+    // Clean the configuration.
+    tablePropsToAdd.put(TABLE_METADATA_PARTITIONS, String.join(",", validPartitionPaths));
+  }
+
+  static List<String> deleteMetadataPartition(HoodieEngineContext context,
+                                              HoodieTableMetaClient mdtMetaClient,
+                                              List<String> metadataPartitions) {
+    List<String> validPartitionPaths = new ArrayList<>();
+
+    for (String partitionPath : metadataPartitions) {
+      if (partitionPath.equals(FUNCTIONAL_INDEX.getPartitionPath())
+          || partitionPath.equals(SECONDARY_INDEX.getPartitionPath())
```

Comment:
Functional index and secondary index uses index definition as the partition path.  Instead of doing exact matches, it's better to check the supported indexes (`FILES`, `COLUMN_STATS`, `BLOOM_FILTERS`, `RECORD_INDEX`) and delete other partitions that do not match these.

Reply:
I used `MetadataPartitionType.fromPartitionPath`, it should work, right?

Reply:
`fromPartitionPath` might be less efficient because it goes through all keys for each partition path look up.